src/bincode/bincode.zig

```diff
@@ -594,9 +672,35 @@ pub fn FieldConfig(comptime T: type) type {
         skip: bool = false,
         default_on_eof: bool = false,
         default_fn: ?fn (alloc: std.mem.Allocator) T = null,
+        skip_write_fn: fn (value: anytype) bool = neverSkip,
+        hashmap: MaybeHashMapConfig(sig.utils.types.hashMapInfo(T)) = .{},
```

Comment:
Isn't `hashmap_config` a bit redundant when it's already the field of a config type?

As for the type, it would say it'd probably be a fair bit harder to realize in the particular way you're suggesting:
```suggestion
        hashmap: ?if (hashMapInfo(T)) |hm_info| struct {
            key: FieldConfig(hm_info.Key) = .{},
            value: FieldConfig(hm_info.Value) = .{}
        } else noreturn = null,
```
it needs to be conditional on the hash map type info regardless, because the `FieldConfig` instantiations will need the key and value types.

The other problem is that it can't just be inlined, because we need to be able to pass `HashMapConfig` to the hash map functions.

As a compromise, here's what getting rid of `MaybeHashMapConfig` and inlining its logic would look like:
```suggestion
        hashmap: if (hashMapInfo) |hm_info| HashMapConfig(hm_info) else struct {} = .{},
```

Reply:
ebddd6d70c796aae3e406ddab297ce9097bf824

lmk wyt

Reply:
well actually that breaks `optional.defaultToNulOnEof`, because it also needs the empty type to be compatible.

Reply:
08ae56850b45b4f1005f9f68837016b306a000a3